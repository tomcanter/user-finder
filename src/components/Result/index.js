// @flow

import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import {Link} from 'react-router-dom';
import Image from 'components/Image';

type Props = {
  username: string,
  avatarUrl: string,
};

function Result({username, avatarUrl}: Props) {
  return (
    <Link
      to={username}
      className={css(styles.Result)}
    >
      <Image
        className={css(styles.Result_img)}
        src={`${avatarUrl}&s=225`}
        alt={username}
        width={140}
        height={140}
      />
      <div className={`${css(styles.Result_userInfo)} userinfo`}>
        <p className={css(styles.Result_username)}>{username}</p>
      </div>
    </Link>
  );
}

const styles = StyleSheet.create({
  Result: {
    position: 'relative',
    display: 'block',

    ':hover .userinfo': {
      opacity: 1,
    },
  },

  Result_img: {
    opacity: 0.8,
    transition: 'opacity .25s ease-in-out',
    // Stop images shifting on hover
    '-webkit-backface-visibility': 'hidden',

    ':hover': {
      opacity: 1,
    },
  },

  Result_username: {
    color: '#fff',
    fontSize: 15,
    fontWeight: 400,
  },

  Result_userInfo: {
    transition: 'opacity .25s ease-in-out',
    opacity: 0,
    backgroundColor: 'rgba(0, 0, 0, .7)',
    bottom: 0,
    left: 0,
    padding: '12px 10px',
    position: 'absolute',
    width: '100%',
  },
});

export default Result;
