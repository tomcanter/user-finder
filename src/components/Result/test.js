import React from 'react';
import renderer from 'react-test-renderer';

import Result from './';

describe('Component: Result', () => {

  it('should render', () => {
    const props = {
      username: 'foo',
      avatarUrl: 'src.jpg',
    };
    const component = renderer.create(
      <Result {...props} />
    );
    expect(component).toMatchSnapshot();
  });

});
