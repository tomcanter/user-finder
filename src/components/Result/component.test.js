import React from 'react';
import renderer from 'react-test-renderer';
import {StyleSheetTestUtils} from 'aphrodite';

import Result from './';

describe('Component: Result', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const props = {
      username: 'foo',
      avatarUrl: 'src.jpg',
    };
    const component = renderer.create(
      <Result {...props} />
    );
    expect(component).toMatchSnapshot();
  });

});
