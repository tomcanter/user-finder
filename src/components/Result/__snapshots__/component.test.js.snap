exports[`Component: Result should render 1`] = `
<div>
  <img
    alt="foo"
    className="Result_img_2jlos-o_O-responsiveImg_kiloh9"
    height={140}
    src="src.jpg"
    width={140} />
</div>
`;
